### ISSUE TITLE
Tapo devices stuck on firmware update when update was not even started at all, incorrect Update state in HA

### ISSUE THREAD

**Issue by GSzabados** [petretiandrea/home-assistant-tapo-p100#747]
## Version of the integration

Latest, 3.0.0, which is actually 3.1.1, but somebody does not update correctly the manifest.jon....

## Configuration

1.3.4

```yaml

```

## Describe the bug

Tapo P100 stuck on firmware update when update was not even started at all

![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/35445496/05050afc-f94b-487c-a41e-aaaa9a4729dc)

## Debug log

```text
This error originated from a custom integration.

Logger: custom_components.tapo.coordinators
Source: helpers/update_coordinator.py:347
integration: TP-Link Tapo (documentation, issues)
First occurred: 14 April 2024 at 23:47:42 (5 occurrences)
Last logged: 12:17:51

Error fetching tapo data: Unexpected exception:

```

**Comment by paulo-serrao** [comment]
same for my P115
![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/35876077/cccf39b2-f98f-4191-9ebf-c40111888e67)

**Comment by GSzabados** [comment]
The same issue with 3.1.2.

**Comment by xenjke** [comment]
Same here!

**Comment by ruipalhinhas** [comment]
Same here, but for a L510 light.

![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/79115158/69407567-9c69-4ea5-802c-037627129409)


```
This error originated from a custom integration.

Logger: custom_components.tapo.coordinators
Source: helpers/update_coordinator.py:347
integration: TP-Link Tapo (documentation, issues)
First occurred: 16 April 2024 at 19:05:55 (6 occurrences)
Last logged: 12:20:29

Error fetching tapo data: Unexpected exception:
```

**Comment by glebrun** [comment]
Same issue for my P110.

**Comment by Game2Freak** [comment]
I have this same issue with my P100

**Comment by LeoXX** [comment]
Same issue

**Comment by fajrstartr** [comment]
Same issue with P110

**Comment by Game2Freak** [comment]
Has anyone tried to update the firmware using the Official Tapo app?

HA is reporting the same firmware as the official app so assuming it’s safe to do the update.

**Comment by glebrun** [comment]
> Has anyone tried to update the firmware using the Official Tapo app?
> 
> HA is reporting the same firmware as the official app so assuming it’s safe to do the update.

Yes, I've just tried that and it seems to have resolved the issue in HA too. I had to hit the "Update Now" button in the Tapo app for the device a few times before it updated. 

I hadn't looked at the Tapo app but it looks like my P110 was set to "Auto Update", but hadn't managed to do so by itself. So maybe HA was simply reporting these attempts by the P110 to update itself? 

tl;dr - update your Tapo device firmware from the Tapo App manually and the notification in HA will go away.

**Comment by GSzabados** [comment]
The issue is that, that it should not show updating. It should show an option to be able to update from HA. As this option should be skippable as well, theough

**Comment by paulo-serrao** [comment]
the way I see it, we should not be forced to update in order for the notification to go away... 

1)it should not display "installing 0%" if we did NOT order it to update thru HA
2)we should be able to disable the notification that a new update is available

**Comment by tymscar** [comment]
I can confirm, same issue here
![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/3742502/429c723a-5e64-4e78-9523-9e852462e87c)
![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/3742502/3c47d3f0-abe0-4640-b981-e425fb868c7d)

**Comment by lucifertfa** [comment]
Tried to add 4 new P100 today, the 3 first got added but the forth got the new update and now i says incorrect credentials (used the same for all 3 previous devices that connected fine) and now also stuck att updating the 3 devices that got connected.

**Comment by GSzabados** [comment]
The issue is a stale, as it has not been still fixed. The issue still persists.

**Comment by tymscar** [comment]
> The issue is a stale, as it has not been still fixed. The issue still persists.

Stalebot is a horrible concept.

**Comment by ruipalhinhas** [comment]
Issue still exists

**Comment by ardejong** [comment]
![image](https://github.com/user-attachments/assets/fc84f58a-3b9c-4c3c-a260-2fdcb21c605e)

same issue here, trying to update via app but downloading 0% remains in screen, after a failed and some devices not updated message

**Comment by Disconn** [comment]
Same Issue

**Comment by GSzabados** [comment]
It was a good run... Today I had to realise that the P100 just died on me. Nothing heavy load or anything, it just did not come back to life when I unplugged and plugged it back, as it was disconnected from the Wifi for days now.

I bought it almost 4 years ago, so it is out of warranty, and to be honest, I wouldn't buy another one based on this, no other plugs have let me down, except the IKEA Tradfri ones... 

I read on Reddit that they are prone to die, or the relay stops working. 

It was a good run, but now I am happy that I don't need to wait anymore that this issue would be fixed....

**Comment by Homas** [comment]
The issue still wasn't fixed and the notifications about the upgrade are annoying

**Comment by tymscar** [comment]
Im having it again now sadly, just like in May. Will try to update through the app again, but honestly it's so much more convenient through Home assistant because I don't really want to have that app installed.

**Comment by ruipalhinhas** [comment]
The problem remains that it shouldn't show up on Home Assistant as being updating with no option to cancel or dismiss the update.
Behind the scenes, it looks like there is an API error in retrieving the update status and it just hangs there (judging by the error message in the log).

**Comment by ruipalhinhas** [comment]
bump to remove stale

**Comment by paulo-serrao** [comment]
Just a FYI, the official [TP-Link Smart Home](https://www.home-assistant.io/integrations/tplink/) integration supports the Tapo plugs: P100, P110, P115, P125M, P135, TP15

I don't think it did then when I first installed this plugin, and I have now migrated into the official integration without any issues.

**Comment by ruipalhinhas** [comment]
> Just a FYI, the official [TP-Link Smart Home](https://www.home-assistant.io/integrations/tplink/) integration supports the Tapo plugs: P100, P110, P115, P125M, P135, TP15
> 
> I don't think it did then when I first installed this plugin, and I have now migrated into the official integration without any issues.

I just tried this for my L510 bulb and it works with the official integration. Thanks for pointing this out!

**Comment by Patriot2407** [comment]
Still an issue

**Comment by psychedelicu** [comment]
Still an issue. Tp-link is trash. Also, TP-Link Smart Home integration not working with P110M.

**Comment by tymscar** [comment]
It works on my P110. How is the M variant different?

**Comment by JackONeill199** [comment]
still an issue

**Comment by numbersixnz** [comment]
still an issue, october 2025.  home assistant 2025.10.2, tapo integration 3.2.1

**Comment by chuegel** [comment]
same issue here with HA 2025.11 3.2.1

**Comment by tymscar** [comment]
I always forget how to fix this and come back to this thread every time. 
I thought to take a look at the code, and the fix is actually a couple of lines only, so I made a PR. If this gets merged, it will all be fine for everyone.

In the meantime, you can use my fork with the fix (make sure to select the correct branch).

I know you changed this to Not Planned @petretiandrea, but does that mean you won't even accept a PR for the fix?

**Comment by Patriot2407** [comment]
Still broken for me
